### PR TITLE
Filter out false positive PyTorch scheduler warning

### LINKED ISSUE #1561: Scheduler warning
### Bug description

When finetuning models, we get this warning:

```
Epoch 1 | iter 1597 step 99 | loss train: 7.316, val: n/a | iter time: 33.55 ms
Epoch 1 | iter 1598 step 99 | loss train: 7.357, val: n/a | iter time: 33.57 ms
Epoch 1 | iter 1599 step 99 | loss train: 7.249, val: n/a | iter time: 33.63 ms
/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/torch/optim/lr_scheduler.py:156: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
Epoch 1 | iter 1600 step 100 | loss train: 7.141, val: n/a | iter time: 35.37 ms (step)
Validating ...
```

 I checked all finetuning codes, and we use `scheduler.step()` without any args. Any ideas where it could come from @Andrei-Aksionov @awaelchli ?

### What operating system are you using?

Linux

### LitGPT Version

``` 

main branch

```

### PR DESCRIPTION
**Pull request by rasbt**
Filters out a false positive warning for PyTorch's scheduler.

Fixes #1561

### CODE REVIEW

File: litgpt/__main__.py [hunk 1]
```diff
@@ -1,5 +1,26 @@
 # Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
 
+import warnings
+import re
+# There is currently a PyTorch bug that raises a false-positive warning.
```
Comment (awaelchli) [inline litgpt/__main__.py:5]: You don't need to put all this in front of the torch import.

File: litgpt/__main__.py [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+# For more information, see https://github.com/Lightning-AI/litgpt/issues/1561
+warning_message = (
```
Comment (awaelchli) [inline litgpt/__main__.py:7]: It's not necessary to have the entire warning message spelled out here. You should be able to just use
```py
"The epoch parameter in `scheduler.step()` was not necessary and is being deprecated.*"
```

File: litgpt/__main__.py [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+    "The epoch parameter in `scheduler.step()` was not necessary and is being deprecated "
+    "where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, "
+    "if epoch is different from None, the closed form is used instead of the new chainable form, "
+    "where available. Please open an issue if you are unable to replicate your use case: "
+    "https://github.com/pytorch/pytorch/issues/new/choose."
+)
+
+
+def custom_filter(warning):
+    return issubclass(warning.category, UserWarning) and re.match(re.escape(warning_message), str(warning.message))
```
Comment (awaelchli) [inline litgpt/__main__.py:17]: Did you try without it? It shouldn't be necessary. The warning is a regular UserWarning

Comment (rasbt) [reply litgpt/__main__.py:17]: Thanks, for some reason the warning still went through with a simpler approach but I am currently working on improving it. Thanks for the feedback here, I am not super familiar with filterwarnings!

**Comment by rasbt** [comment]
Thanks @awaelchli ! I simplified this now and it seems to work!

**Review by awaelchli — APPROVED** [review]
